compiler/IREmitter/Payload/patches/hash.c

```diff
@@ -36,3 +37,30 @@ VALUE sorbet_rb_hash_to_h(VALUE hash) {
     }
     return hash;
 }
+
+// no-block rb_hash_fetch_m: https://github.com/ruby/ruby/blob/ruby_2_7/hash.c#L2113-L2147
+VALUE sorbet_rb_hash_fetch_m(VALUE recv, ID fun, int argc, const VALUE *const restrict argv, BlockFFIType blk,
```

Comment:
Note that the `withBlock` version of this is not hard to implement either, just wanted to limit the scope of this PR to address the immediate bug.